spec/controllers/prompts_controller_spec.rb

```diff
@@ -1,18 +1,126 @@
-require 'spec_helper'
+require "spec_helper"
 
 describe PromptsController do
   include LoginMacros
+  include RedirectExpectationHelper
+  let(:collection) { create(:collection) }
+  let(:open_signup) do
+    signups = create(:challenge_signup)
+    signups.collection.challenge.signup_open = true
+    signups.collection.challenge.save
+    signups
+  end
+  let(:signup) { create(:challenge_signup) }
 
   before do
     fake_login
   end
 
-  describe 'no_prompt' do
-    it 'should show an error and redirect' do
-      signups = create(:challenge_signup)
-      post :edit, collection_id: signups.collection.name
-      expect(response).to redirect_to(collection_path(signups.collection))
-      expect(flash[:error]).to eq "What prompt did you want to work on?"
+  describe "no_challenge" do
+    it "should show an error and redirect" do
+      get :show, collection_id: collection.name
+      it_redirects_to_with_error(collection_path(collection), "What challenge did you want to sign up for?")
+    end
+  end
+
+  describe "no_signup" do
+    it "should show an error and redirect" do
+      post :create, collection_id: signup.collection.name
+      it_redirects_to_with_error(collection_signups_path(signup.collection) + "/new",\
+                                 "Please submit a basic sign-up with the required fields first.")
+    end
+  end
+
+  describe "signups_closed" do
+    it "should show an error and redirect" do
+      # Login as the signup owner
+      fake_login_known_user(Pseud.find(ChallengeSignup.in_collection(signup.collection).first.pseud_id).user)
```

Comment:
Yeah, so on line 15 instead of

```
 before do
    fake_login
  end
```

have something like

```
let(:user) { create(:user) }
before do
  fake_login_known_user(user)
end
```

Then have the `signups_closed` context look something like this:

```
describe "signups_closed" do
  let(:user) { Pseud.find(ChallengeSignup.in_collection(signup.collection).first.pseud_id).user }
  it "should show an error and redirect" do
    post :create, collection_id: signup.collection.name
    it_redirects_to_with_error(signup.collection, "Signup is currently closed: please contact a moderator for help.")
  end
end
```

And do that for every context where you want to log in as the signup owner, or some other user other than the one created up above in the larger context.